lib/maths/CBoostedTreeFactory.cc

```diff
@@ -120,10 +135,22 @@ CBoostedTreeFactory::restoreFor(core::CDataFrame& frame, std::size_t dependentVa
         return nullptr;
     }
 
+    switch (m_TreeImpl->m_InitializationStage) {
+    case CBoostedTreeImpl::E_FullyInitialized:
+        break;
+    case CBoostedTreeImpl::E_NotInitialized:
+    case CBoostedTreeImpl::E_SoftTreeDepthLimitInitialized:
+    case CBoostedTreeImpl::E_DepthPenaltyMultiplierInitialized:
+    case CBoostedTreeImpl::E_TreeSizePenaltyMultiplierInitialized:
+    case CBoostedTreeImpl::E_LeafWeightPenaltyMultiplierInitialized:
+    case CBoostedTreeImpl::E_DownsampleFactorInitialized:
+    case CBoostedTreeImpl::E_EtaInitialized:
```

Comment:
Are you actually setting all these stage values somewhere?

Reply:
As below, this happens in `skipCheckpointIfAtOrAfter`.